common/llvm.saw

```diff
@@ -14,13 +28,15 @@ let llvm_verify module function_name overrides spec =
 
 // Declare constants used to denote whether an allocation is
 // 'READONLY' (CONST) or not (WRITE).
-let CONST = {{ False }};
-let WRITE = {{ True }};
+let CONST  = {{ 0 : [2] }}; //Points to allocated space that is READ ONLY.
+let WRITE  = {{ 1 : [2] }}; //Points to allocated space that can be modified.
+let UNINIT = {{ 2 : [2] }}; //Pointer does not point to allocated space.
 
 // Returns a newly-allocated pointer of the given type.
-let pointer (rw : Term) (type : LLVMType) = if (eval_bool {{ rw == WRITE }})
-                      then crucible_alloc type
-                      else crucible_alloc_readonly type;
+let pointer (rw : Term) (type : LLVMType) =
+  if     (eval_bool {{ rw == 0 }}) then crucible_alloc_readonly type
+  else if(eval_bool {{ rw == 1 }}) then crucible_alloc type
+  else                                  crucible_fresh_pointer type;
```

Comment:
I added another input `UNINT` that can be used to create an uninitialized pointer or one which points to NULL.

Reply:
I think passing parameters is more flexible than having lots of different functions for creating different types of pointers. I've seen benefit already from having the read-only parameter passed in, most notably when creating functions that return structs.

Reply:
That argument makes sense to me. I'd call the 3rd case `UNALLOC` instead of `UNINIT`, though. All three are uninitialized.

Reply:
I think you're right. The pointer either points to something that is readonly, writable, or unallocated/NULL.